src/main/java/duke/util/DukeDB.java

```diff
@@ -0,0 +1,168 @@
+                        return x;
+                    })
+                    .orElseGet(() -> {
+                        Duke.printMsg("Unable to create the file. Check if the file already exists. Defaulting to " + "default database");
+                        this.databasePath = "./data/dukeStore.txt";
+                        return null;
+                    });
+        } else {
+            this.databasePath = "./data/dukeStore.txt";
+        }
+
+    }
+
+    @Deprecated
+    public static void drop() {
+        File file = new File("./data/dukeStore.txt");
+        long size = file.getTotalSpace();
+        boolean deleted = file.delete();
+        if (deleted) {
+            Duke.printMsg("Successfully dropped database of " + size + "B.");
+        } else {
+            Duke.printMsg("Error deleting file.");
+        }
+    }
+
+    public void save(ArrayList<Task> arr) {
+        try {
+            FileWriter writer = new FileWriter(this.databasePath);
+            for (Task item : arr) {
+                writer.write(item.saveString());
+                writer.write("\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            Duke.printMsg(e.toString() + " Error saving task");
+        }
+    }
+
+    public Optional<ArrayList<Task>> load() {
+        try {
+            ArrayList<Task> arr = new ArrayList<>();
+            File database = new File(databasePath);
+            Scanner reader = new Scanner(database);
+            while (reader.hasNextLine()) {
+                String strTask = reader.nextLine();
+                String[] splitTask = strTask.split("\\|");
+                Optional<Task> task = this.readTask(splitTask);
+                task.ifPresentOrElse(arr::add,
+                        () -> Duke.printMsg("Error loading task. Database is corrupt."));
+            }
+            return Optional.of(arr);
+        } catch (FileNotFoundException e) {
+            Duke.printMsg(e.toString());
+        } catch (Error f) {
+            Duke.printMsg("Problem reading the database format. Was the database modified externally?");
+        }
+        return Optional.empty();
+    }
+
+    public Optional<Boolean> create() {
+        File file = new File("./data/dukeStore.txt");
+        try {
+            if (!file.exists()) {
+                file.getParentFile()
+                        .mkdirs();
+                return Optional.of(file.createNewFile());
+            } else {
+                return Optional.empty();
+            }
+        } catch (IOException e) {
+            return Optional.empty();
+        }
+    }
+
+    private Optional<Task> readTodo(String[] splitCommand) {
+        if (splitCommand.length != 3) {
+            return Optional.empty();
+        } else {
+            String description = splitCommand[1];
+            boolean isDone = splitCommand[2].equals("1");
+            return Optional.of(new Todo(description,
+                    isDone));
```

Comment:
Not sure if the indentation of the second line should be more? Maybe align with the first parameter?